lib/Federation/FederationManager.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2021, Gary Kim <gary@garykim.dev>
+ *
+ * @author Gary Kim <gary@garykim.dev>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Federation;
+
+use OCA\Talk\Exceptions\UnauthorizedException;
+use OCA\Talk\Room;
+use OCP\DB\Exception as DBException;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\IConfig;
+use OCP\IDBConnection;
+use OCP\IUser;
+
+class FederationManager {
+	/** @var IDBConnection */
+	private $db;
+	/** @var IConfig */
+	private $config;
+
+	public function __construct (
+		IDBConnection $db,
+		IConfig $config
+	) {
+		$this->db = $db;
+		$this->config = $config;
+	}
+
+	/**
+	 * Determine if Talk federation is enabled on this instance
+	 * @return bool
+	 */
+	public function isEnabled(): bool {
+		// TODO: Set to true once implementation is complete
+		return $this->config->getSystemValueBool('talk_federation_enabled', false);
+	}
+
+	/**
+	 * @param IUser $user
+	 * @param string $roomToken
+	 * @param string $remoteUrl
+	 * @param string $sharedSecret
+	 * @return int share id for this specific remote room share
+	 * @throws DBException
+	 */
+	public function addRemoteRoom(IUser $user, string $roomToken, string $remoteUrl, string $sharedSecret): int {
+		$qb = $this->db->getQueryBuilder();
+
+		$query = $qb->insert('talk_rooms_external')
```

Comment:
Use a Mapper and a Model like lib/Model/AttendeeMapper.php and lib/Model/Attendee.php